solana-programs/anchor/audius-data/programs/audius-data/src/lib.rs

```diff
@@ -109,11 +110,13 @@ pub mod audius_data {
     */
     pub fn create_track(ctx: Context<CreateTrack>, metadata: String) -> ProgramResult {
         msg!("Audius::CreateTrack");
-        if ctx.accounts.authority.key() != ctx.accounts.user.authority.key() {
+        if ctx.accounts.authority.key() != ctx.accounts.user.authority {
             return Err(ErrorCode::Unauthorized.into());
         }
         // Set owner to user storage account
         ctx.accounts.track.owner = ctx.accounts.user.key();
+        ctx.accounts.track.track_id = ctx.accounts.audius_admin.track_id;
+        ctx.accounts.audius_admin.track_id = ctx.accounts.audius_admin.track_id + 1;
```

Comment:
Comment on what this line is doing would be nice (obviously it's incrementing the admin id), but why